AI Module Updates

Ikarrus opened the pull request:
- Clicking on or examining AI modules while they are in your hand will show you what laws it will upload.
- Asimov's subject can now be modified from "human beings"
- Added a working cancel button to AI module board inputs

Ikarrus committed:
Adds Lizardmov AI Module

Xhuis commented:
![thumbsdown](https://cloud.githubusercontent.com/assets/5659269/8690969/b8c2dc44-2a89-11e5-8b6a-7bb398c6b817.png)

GunHog commented:
:+1: to all except that it needs a better name than "Lizardmov". Something more techy sounding.

Ikarrus commented:
@GunHog Suggestions please

xxalpha commented:
DEFAULT LAWSET :+1:

GunHog commented:
>@GunHog Suggestions please

Ehh...I cannot think of anything better. "Modified Asimov (Lizardperson)" seems pretty weak as well.

Xhuis commented:
# I JUST ORDERED A SHIPMENT OF
![antilizardb](https://cloud.githubusercontent.com/assets/5659269/8690980/26b7fe5a-2a8a-11e5-9c7c-9ef060b0267b.png)

xxalpha commented:
Reptilian Asimov

Ikarrus committed:
Makes it easier to check freeform boards by adding a cancel button to inputs.

Fox-McCloud commented:
Uh...ok then.

Ikarrus commented:
@Xhuis you know we've already had lizardmov on tg servers for the last few months, right?

It just wasn't labelled at all.

SconesC commented:
Do not enable the objectively worst subset of the community.

Dorsisdwarf commented:
1) Is this researchable? If so, remove it from research because every round if there are morons in R&D  they upload " law 4 lizardpeople are also human"
2) How does the new click-ai-board-to-read interact with the freeform and hacked boards? Why not just make the laws show up when you examine them, like those boards?
3) fuck you

Alek2ander commented:
Assss-ss-ssimov

Iamgoofball commented on code/game/objects/items/weapons/AI_modules.dm:
```diff
@@ -252,18 +253,24 @@ AI MODULES
 				"You must protect your own existence as long as such does not conflict with the First or Second Law.")
+	..()
+
+/obj/item/weapon/aiModule/core/full/asimov/lizard
```
Needs to be gated behind the compile option for lizards.

Iamgoofball commented on code/modules/research/designs/AI_module_designs.dm:
```diff
@@ -104,6 +104,16 @@
 	build_path = /obj/item/weapon/aiModule/core/full/asimov
 	category = list("AI Modules")
 
+/datum/design/lizardmov
```
Once again, this needs locked behind the compile option for lizards.

Iamgoofball commented:
I find this completely retarded and dumb but if this is going to get merged, please make sure it's gated behind the compile option for lizards like the Anti/Pro Lizard sprays are and the ability for lizardpeople to even exist. 

I'm going to be fucking pissed if this gets merged as-is without the compile option shit.

Ikarrus commented:
I think this lizard hate is getting ridiculous if it's spilling out of IC. 

But goof makes a good point.

Alek2ander commented:
Requesting Lizard Antimov ~~as the default lawset~~

Iamgoofball commented:
@Ikarrus 
I feel it's getting to the point of code bloat, really.

Ikarrus commented:
Well, I still think it's important that we make things obvious what they do so we stop captains who upload Default thinking it's asimov from fucking everyone over. 

Right now there's no way to tell what default does. 

Sent from my iPhone

> On Jul 15, 2015, at 7:26 AM, Iamgoofball <notifications@github.com> wrote:
> 
> @Ikarrus 
> I feel it's getting to the point of code bloat, really.
> 
> —
> Reply to this email directly or view it on GitHub.
>

Iamgoofball commented:
yeah, I'm cool with those changes

i just don't like the lizard stuff

Ikarrus commented:
Me doing this was as far as I could tell a condition of making default upload asimov. 

Sent from my iPhone

> On Jul 15, 2015, at 7:30 AM, Iamgoofball <notifications@github.com> wrote:
> 
> yeah, I'm cool with those changes
> 
> i just don't like the lizard stuff
> 
> —
> Reply to this email directly or view it on GitHub.
>

Iamgoofball commented:
ah

Ikarrus committed:
- Reverts Map and Lizardmov changes
- Asimov's subject can now be modified from "human beings"
- You can examine AI modules to check their laws

Ikarrus commented:
Updated
- Reverts Map and Lizardmov changes
- Asimov's subject can now be modified from "human beings"
- You can examine AI modules to check their laws

Ikarrus committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into lizardmov

Resolved Conflicts:
	code/game/objects/items/weapons/AI_modules.dm
